### PR TITLE
#19 #39 Documentation of /Room namespace

### PR DESCRIPTION
**Pull request by michaelKurowski**
**Reference to related ticket:**
#19 #39
**Description of changes:**
- New command `npm run generate-docs` which creates docs in `/out` directory
- Docs for `/Room` web socket namespace
- Room API redesign
- Ignoring `/out` directory in server directory which will now be used to store generated docs

### CODE REVIEW

**Comment by michaelKurowski** [comment]
@Jordan141 You might want to track progress on this PR.

**Comment by michaelKurowski** [comment]
TODO: 

- Move uuid type definition somewhere else
- Adjust tests to new API

**Comment by michaelKurowski** [comment]
TODO:
Adjust tests to new API
Add docs for connection, disconnected and connected

**Comment by michaelKurowski** [comment]
TODO: Add docs for connection, disconnected and connected

**Comment by michaelKurowski** [comment]
@Jordan141 Check from API design POV.
@MarcinGrzeszczak Check from code POV.

File: src/server/ws-routes/controllers/Room.js [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
 
-	static message(data, socket, connections) {
+	/**
+	 * User sends a message
+	 * @name message
+	 * @memberof Room
+	 * @member
+	 * @property {module:dataTypes.uuid} roomId Room to receive a message
+	 * @property {string} message Text message
+	 * @property {string} username Username of user that joined (only for server-sourced emits)
```
Comment (Jordan141) [inline src/server/ws-routes/controllers/Room.js]: Duplicate description

File: src/server/ws-routes/controllers/Room.js [hunk 2]
```diff
@@ -1,35 +1,79 @@
...
 	}
 
-	static leave(data, socket, connections) {
+	/**
+	 * User leaves a room
+	 * @name leave
+	 * @memberof Room
+	 * @member
+	 * @property {module:dataTypes.uuid} roomId Room to leave
+	 * @property {string} username Username of user that joined (only for server-sourced emits)
```
Comment (Jordan141) [inline src/server/ws-routes/controllers/Room.js]: Duplicate description here too

**Comment by michaelKurowski** [comment]
@Jordan141 Wrong descriptions for `username` property in `Room` namespace events fixed.

**Comment by michaelKurowski** [comment]
@Jordan141 This PR will also resolve this ticket: #39

**Pull request merged**